SMCKit/SMC.swift

```diff
@@ -995,18 +1068,115 @@ public struct SMC {
+                i++
+                let indexResult = readSMCByIndex(i)
+                if indexResult.IOReturn != kIOReturnSuccess {
+                        continue
+                }
+                returnDictionary[SMC.decodeSMCKey(indexResult.key)] = (decodeType(indexResult.data, dataType: indexResult.dataType, dataSize: indexResult.dataSize), SMC.decodeSMCKey(indexResult.dataType))
+                //NSLog("%@", SMC.decodeSMCKey(indexResult.dataType))
+            }
+            
+            return (returnDictionary, result.IOReturn, result.kSMC)
+    }
 
 
     //--------------------------------------------------------------------------
     // MARK: PRIVATE METHODS
     //--------------------------------------------------------------------------
+    
+    private func decodeType(data : [UInt8], dataType : UInt32, dataSize : IOByteCount) -> Any? {
+        if (SMC.decodeSMCKey(dataType) == "flag") {
+            return (data[0] == 1) ? true : false;
+        } else if (SMC.decodeSMCKey(dataType) == "{pwm") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) * 100 / 65536.0
+        } else if (SMC.decodeSMCKey(dataType) == "fp1f") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) / 32768.0
+        } else if (SMC.decodeSMCKey(dataType) == "fp4c") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) / 4096.0
+        } else if (SMC.decodeSMCKey(dataType) == "fp5b") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) / 2048.0
+        } else if (SMC.decodeSMCKey(dataType) == "fp6a") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) / 1024.0
+        } else if (SMC.decodeSMCKey(dataType) == "fp79") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) / 512.0
+        } else if (SMC.decodeSMCKey(dataType) == "fp88") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) / 256.0
+        } else if (SMC.decodeSMCKey(dataType) == "fpa6") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) / 64.0
+        } else if (SMC.decodeSMCKey(dataType) == "fpc4") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) / 16.0
+        } else if (SMC.decodeSMCKey(dataType) == "fpe2") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) / 4.0
+        } else if (SMC.decodeSMCKey(dataType) == "sp1e") {
+            return Float(ntohs(UnsafePointer<UInt16>(data).memory)) / 16384.0
+        } else if (SMC.decodeSMCKey(dataType) == "ch8*") {
+            let bytes = NSData(bytes: data, length: sizeofValue(data))
+            let str = NSString(data: bytes, encoding: NSASCIIStringEncoding)
+            return str as? String
+        } else if (SMC.decodeSMCKey(dataType) == "sp3c") {
+            return Double(data[0]) + Double(data[1]/10)
+        } else {
+            return nil
+        }
 
+        /*case SMC.encodeSMCKey(SMC.DataType.SP4B.rawValue):
```

Comment:
Commented code like this should be removed. Saw a few NSLog lines commented out as well.